static/panes/ast-view.js

```diff
@@ -110,12 +118,16 @@ Ast.prototype.onCompileResult = function (id, compiler, result, lang) {
         this.showAstResults(result.astOutput);
     }
     else if (compiler.supportsAstView) {
-        this.showAstResults('<No output>');
+        this.showAstResults([{text: '<No output>'}]);
     }
 
     if (lang && lang.monaco && this.getCurrentEditorLanguage() !== lang.monaco) {
         monaco.editor.setModelLanguage(this.astEditor.getModel(), lang.monaco);
     }
+
+    // Why call this explicitly instead of just listening to the "colours" event?
+    // Because the recolouring happens before this editors value is set using "showIrResults".
+    this.onColours(this._compilerid, this.lastColours, this.lastColourScheme);
```

Comment:
Darn it re: colours event there. I would worry there's some edge cases but...I can't think of any. The colours event comes late as in theory all the different colour contributors should agree on the same colour for the same lines. It's been a while since I looked at the code though, and I suspect your understanding of what's going on is much better than my foggy memory.

Reply:
Frankly, I understand it only superficially. I've copied this part entirely from
https://github.com/compiler-explorer/compiler-explorer/blob/main/static/panes/ir-view.js#L156
I'll add a comment saying this.
However, I tried removing this line or the binding to the event on [l.89](https://github.com/compiler-explorer/compiler-explorer/pull/2511/files#diff-b7db153f9bd10df795f14e5b3e97f7c010aa049cccbcd8414958c582150bce63R89): if any of them is absent the AST view remains colourless.

Reply:
Got it...I'll look in to it...

Reply:
(filed as compiler-explorer/compiler-explorer#2513 for reference )